### PR TITLE
Use select in withDispatch to avoid using unused props

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Partially addresses #12890.

With the changes introduced by #11851, we seek to refactor components which provide data through `withSelect` for the sole purpose of satisfying a dispatch callback. These components instead should use the new registry argument. This PR covers the following:
- [x] `EditorGlobalKeyboardShortcuts` (the `isDirty` prop)
- [x] `InserterMenu` (the `selectedBlock` prop)
- [x] `EditorModeKeyboardShortcuts` (the `hasBlockSelection` prop)

## How has this been tested?
`npm test` & `npm run test-e2e`

* Make sure that inserter still works.
* Make sure keyboard shortcuts still work.

## Types of changes
Refactoring.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Review by aduth — APPROVED** [review]

File: packages/editor/src/components/inserter/menu.js [hunk 1]
```diff
@@ -359,14 +357,12 @@ export default compose(
 		const rootBlockName = getBlockName( rootClientId );
 
 		return {
-			selectedBlock: getSelectedBlock(),
 			rootChildBlocks: getChildBlockNames( rootBlockName ),
-			title: getEditedPostAttribute( 'title' ),
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/editor/src/components/inserter/menu.js]: Yes, I forgot to comment. It's a dead code.

[the remark this answers is not in the document]
Comment (aduth) [reply packages/editor/src/components/inserter/menu.js]: I wish we could lint this 😄

File: packages/edit-post/src/components/keyboard-shortcuts/index.js [hunk 2]
```diff
@@ -58,14 +58,14 @@ export default compose( [
...
 	} ) ),
-	withDispatch( ( dispatch, { hasBlockSelection } ) => ( {
+	withDispatch( ( dispatch, ownProps, { select } ) => ( {
 		switchMode( mode ) {
 			dispatch( 'core/edit-post' ).switchEditorMode( mode );
 		},
 		openSidebar() {
-			const sidebarToOpen = hasBlockSelection ? 'edit-post/block' : 'edit-post/document';
+			const { getBlockSelectionStart } = select( 'core/editor' );
+			const sidebarToOpen = !! getBlockSelectionStart() ? 'edit-post/block' : 'edit-post/document';
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/edit-post/src/components/keyboard-shortcuts/index.js]: Updated.

File: packages/editor/src/components/editor-global-keyboard-shortcuts/index.js [hunk 3]
```diff
@@ -186,10 +184,11 @@ export default compose( [
 				// TODO: This should be handled in the `savePost` effect in
 				// considering `isSaveable`. See note on `isEditedPostSaveable`
 				// selector about dirtiness and meta-boxes. When removing, also
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/editor/src/components/editor-global-keyboard-shortcuts/index.js]: Removed.

**Pull request merged**